### PR TITLE
Convertion of GDO screen to EGO

### PR DESCRIPTION
**Pull request by Ordinastie**
SGWindow is a base for windows and panel with SGCraft looks.
Builders are provided for both the background and the component for easy configurable result

Fixed the invisible cursor bug by scheduling the opening of the gui on the client thread.
Most of the data sent to the client is not necessary anymore and should be refatored away, both in the packet and the item right click.
Fixed the method to find a nearby gate. I recommend changing all the calls to the old method everywhere else.

I left the .editorconfig that allows the formatting to be directory dependant. It's handled by IDEA natively but Eclipse might need a plugin.

### CODE REVIEW

File: src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    public GuiIcon gateStatus() {
+        if (gate == null) {
+            return GuiIcon.NONE;
+        }
+
+        return gateTextures.computeIfAbsent(gateStatusLabel(), s -> GuiIcon.full(new GuiTexture(new ResourceLocation("sgcraft:" + s))));
+    }
+
+    private String gateStatusLabel() {
+        String status = "textures/";
```
Comment (Zidane) [inline src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java:99]: StringBuilder, you are creating 5 String objects per call here.

File: src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    private void disconnectGate() {
+        GdoNetworkHandler.sendGdoInputToServer(gate, 3);
+    }
+
+    public static GateControlComponentBuilder builder(SGBaseTE gate, boolean local) {
+        return new GateControlComponentBuilder(gate, local);
+    }
+
+    public static class GateControlComponentBuilder extends UIContainerBuilderG<GateControlComponentBuilder, GateControlComponent> {
```
Comment (Zidane) [inline src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java:142]: This makes the class lines needlessly long for these.

```java
GateControlComponent.GateControlComponentBuilder builder...
```

These should all be called Builder, so it becomes clean:

```java
GateControlComponent.Builder builder = ...
```

File: src/mod/gcewing/sg/util/GateUtil.java [hunk 3]
```diff
@@ -2,20 +2,41 @@
...
 import gcewing.sg.tileentity.DHDTE;
 import gcewing.sg.tileentity.SGBaseTE;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
 public class GateUtil {
+
+    public static SGBaseTE findGate(World world, EntityPlayer player, int radius) {
```
Comment (Zidane) [inline src/mod/gcewing/sg/util/GateUtil.java:12]: This should not require a player. Should take the blockpos.

File: src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java (hunk 1 above)
Comment (Ordinastie) [reply src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java:99]: I started using StringBuilder, but IDEA complained and told me I should just concatenate strings...

File: src/mod/gcewing/sg/util/GateUtil.java (hunk 3 above)
Comment (Ordinastie) [reply src/mod/gcewing/sg/util/GateUtil.java:12]: findGate is never used with a BlockPos starting point, it always comes from a player. Everytime, the blockPos had to be constructed from the player.
I would have provided the player param as an overload if at least one place in the code started from a block pos

Comment (Zidane) [reply src/mod/gcewing/sg/util/GateUtil.java:12]: That doesn't matter. It does not use the Player at all BESIDES getting the block pos.

File: src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java (hunk 1 above)
Comment (Zidane) [reply src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java:99]: IDEA is retarded then.

It is relying on JIT to inline and optimize the call away. I prefer being 100% sure it did by not creating the objects to begin with.

**Pull request closed without merging**